src/trace/trace.cpp

```diff
@@ -332,42 +347,53 @@ void Trace::add_lo2s_property(const std::string& name, const std::string& value)
 otf2::writer::local& Trace::sample_writer(pid_t pid, pid_t tid)
 {
     auto name = (boost::format("thread %d") % tid).str();
-    auto location =
-        locations_.emplace(location_ref(), intern(name), location_groups_process_.at(pid),
-                           otf2::definition::location::location_type::cpu_thread);
-    process_comm_groups_.at(pid).add_member(location);
-    return archive()(location);
-}
 
+    // As the tid is unique in this context, create only one writer/location per tid
+    auto location = thread_locations_.emplace(
+        std::piecewise_construct, std::forward_as_tuple(tid),
+        std::forward_as_tuple(location_ref(), intern(name), location_groups_process_.at(pid),
+                              otf2::definition::location::location_type::cpu_thread));
+    process_comm_groups_.at(pid).add_member(location.first->second);
+
+    return archive()(location.first->second);
+}
 otf2::writer::local& Trace::cpu_writer(int cpuid)
 {
-    auto name = intern((boost::format("cpu %d") % cpuid).str());
+    auto name = (boost::format("cpu %d") % cpuid).str();
```

Comment:
Maybe I was a bit unclear, but you got me wrong on this one. I actually meant to use the old code snippet, which does the `intern()` call once. While the current code should work fine 
as `intern()` does caching, it's less efficient than reusing the interned string:

```
auto name = intern((boost::format("cpu %d") % cpuid).str());
```